contracts/JoinFactory.sol

```diff
@@ -63,7 +18,7 @@ contract JoinFactory is IJoinFactory {
   /// @return join The join address.
   function createJoin(address asset) external override returns (address) {
     _nextAsset = asset;
-    Join join = new Join{salt: keccak256(abi.encodePacked(asset))}();
+    Join join = new Join();
```

Comment:
Please change the `Join` constructor to accept the `asset` as a parameter, we don't need to use a callback anymore because we don't use CREATE2 anymore.